tests/unit_tests/test_subtensor.py

```diff
@@ -275,90 +249,62 @@ def test_determine_chain_endpoint_and_network(
     """Tests when query_subtensor returns None."""
-    subtensor.subnet_exists = mocker.MagicMock(return_value=True)
-    subtensor.query_subtensor = mocker.MagicMock(return_value=None)
-    assert subtensor._get_hyperparameter("Difficulty", 1, None) is None
+    subtensor.subnet_exists = mocker.AsyncMock(return_value=True)
+    subtensor.query_subtensor = mocker.AsyncMock(return_value=None)
+    assert await subtensor._get_hyperparameter("Difficulty", 1, None) is None
     subtensor.subnet_exists.assert_called_once_with(1, None)
     subtensor.query_subtensor.assert_called_once_with("Difficulty", None, [1])
 
 
-def test_hyperparameter_result_has_no_value(subtensor, mocker):
+@pytest.mark.asyncio
+async def test_hyperparameter_result_has_no_value(subtensor, mocker):
     """Test when the result has no 'value' attribute."""
 
-    subtensor.subnet_exists = mocker.MagicMock(return_value=True)
-    subtensor.query_subtensor = mocker.MagicMock(return_value=None)
-    assert subtensor._get_hyperparameter("Difficulty", 1, None) is None
+    subtensor.subnet_exists = mocker.AsyncMock(return_value=True)
+    subtensor.query_subtensor = mocker.AsyncMock(return_value=None)
+    assert await subtensor._get_hyperparameter("Difficulty", 1, None) is None
     subtensor.subnet_exists.assert_called_once_with(1, None)
     subtensor.query_subtensor.assert_called_once_with("Difficulty", None, [1])
 
 
-def test_hyperparameter_success_int(subtensor, mocker):
+@pytest.mark.asyncio
+async def test_hyperparameter_success_int(subtensor, mocker):
     """Test when query_subtensor returns an integer value."""
-    subtensor.subnet_exists = mocker.MagicMock(return_value=True)
-    subtensor.query_subtensor = mocker.MagicMock(
+    subtensor.subnet_exists = mocker.AsyncMock(return_value=True)
+    subtensor.query_subtensor = mocker.AsyncMock(
         return_value=mocker.MagicMock(value=100)
     )
-    assert subtensor._get_hyperparameter("Difficulty", 1, None) == 100
+    assert await subtensor._get_hyperparameter("Difficulty", 1, None) == 100
     subtensor.subnet_exists.assert_called_once_with(1, None)
     subtensor.query_subtensor.assert_called_once_with("Difficulty", None, [1])
 
 
-def test_hyperparameter_success_float(subtensor, mocker):
+@pytest.mark.asyncio
+async def test_hyperparameter_success_float(subtensor, mocker):
     """Test when query_subtensor returns a float value."""
-    subtensor.subnet_exists = mocker.MagicMock(return_value=True)
-    subtensor.query_subtensor = mocker.MagicMock(
+    subtensor.subnet_exists = mocker.AsyncMock(return_value=True)
+    subtensor.query_subtensor = mocker.AsyncMock(
         return_value=mocker.MagicMock(value=0.5)
     )
-    assert subtensor._get_hyperparameter("Difficulty", 1, None) == 0.5
+    assert await subtensor._get_hyperparameter("Difficulty", 1, None) == 0.5
     subtensor.subnet_exists.assert_called_once_with(1, None)
     subtensor.query_subtensor.assert_called_once_with("Difficulty", None, [1])
```

Comment:
Do these not need to be awaited? They might not, I just don't know.

Reply:
in assertions no, because it is checking the returned object such is sync already